### PR TITLE
st1 [1782][IMP] account_invoice_validate_send_email:accept tracking_url blank

### PR DESCRIPTION
**Pull request by TomohiroAbe-Quartile**
[1782](https://www.quartile.co/web#id=1782&action=771&model=project.task&view_type=form&menu_id=505)

### CODE REVIEW

**Comment by TomohiroAbe-Quartile** [comment]
Probably it because some carrier_info models did not include tracking_url.

Pre-commit failed with line below:

`    RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.6'
`

File: account_invoice_validate_send_email/models/account_invoice.py [hunk 1]
```diff
@@ -125,10 +125,10 @@ def _compute_carrier_info(self):
...
             carrier_recs = invoice.picking_ids.mapped("carrier_info_id")
             if carrier_recs:
-                invoice.carrier_info_name = ", ".join(x.name for x in carrier_recs)
-                invoice.carrier_tracking_url = ", ".join(
-                    x.tracking_url for x in carrier_recs
-                )
+                for x in carrier_recs:
+                    invoice.carrier_info_name = ", ".join(x.name)
+                    if x.tracking_url:
+                        invoice.carrier_tracking_url = ", ".join(x.tracking_url)
```
Comment (yostashiro) [inline account_invoice_validate_send_email/models/account_invoice.py]:
```suggestion
                invoice.carrier_info_name = ", ".join(x.name for x in carrier_recs)
                tracking_urls = carrier_recs.mapped("tracking_url")
                if tracking_urls:
                    invoice.carrier_tracking_url = ", ".join(url for url in tracking_urls if url)
```

@TomohiroAbe-Quartile Can you check if above makes sense.  Haven't tested it myself.

Comment (TomohiroAbe-Quartile) [reply account_invoice_validate_send_email/models/account_invoice.py]: @yostashiro 
I checked and it works as expected.
Situation like below:
1 invoice record has 2 deliveries.
1 delivery has carrier_name and carrier_number.
And other one has carrier_name, but not has carrier_number.

And succeed to validate, and send e-mail.

**Review by yostashiro — APPROVED** [review]
👍

**Pull request merged**